kernel/source/interrupt/irq.c

```diff
@@ -155,8 +244,9 @@ void IrqHandler(ThreadContext* context)
 	}
 	if(flags & IRQF_IPC) {
 		//gecko_printf("IRQ: IPC\n");
-		ipc_irq();
+		EnqueueEventHandler(IRQ_IPC);
 		write32(HW_ARMIRQFLAG, IRQF_IPC);
+		clear32(HW_ARMIRQMASK, IRQF_IPC);
```

Comment:
ios seems to do the clear first, followed the write and enqueue. i suppose its to shut it up asap haha

Reply:
fixed